sqlglot/dialects/presto.py

```diff
@@ -118,6 +119,11 @@ class Presto(Dialect):
     time_format = "'%Y-%m-%d %H:%i:%S'"
     time_mapping = MySQL.time_mapping
 
+    class Tokenizer(Tokenizer):
+        KEYWORDS = {
+            k: v for k, v in Tokenizer.KEYWORDS.items() if v != TokenType.FILTER
```

Comment:
i don’t think this is right, you just need to add filter to the list of functions in parser

Reply:
filter is used i agg expressions like sum(x) filter…

Reply:
I didn't think that was supported by Spark but I just confirmed it does actually work. TIL. Well the bad news is that it looks like sqlglot evaluates the keywords before the functions so when I don't have that included it tries to parse FILTER within the context of an aggregate instead of a function.

Reply:
I can't get this syntax to work on our Spark release:
```
SELECT
  SUM(play_secs) FILTER (where account_id > 1)
FROM
  VALUES(1, 2) tab(account_id, play_secs)
```
mismatched input '(' expecting <EOF>(line 2, pos 24)

== SQL ==
SELECT
  SUM(play_secs) FILTER (where account_id > 1)
------------------------^^^
FROM
  VALUES(1, 2) tab(account_id, play_secs)
```

Reply:
If I remove the that keyword override then it doesn't work since it looks like sqlglot treats it as a filter like in the context above instead of the custom function I am defining.

Reply:
I guess regardless if it works for our spark it does seem to be supported: https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-groupby.html#description

So we need to figure this out regardless.

Reply:
i’ll figure it out tonight